packages/electron-app/src/database/connection.js

```
@@ -96,23 +127,134 @@ async function backupDatabase() {
+      message: 'Database successfully restored from backup. All data has been replaced with the backup data.' 
+    };
+    
   } catch (error) {
-    return { success: false, message: error instanceof Error ? error.message : String(error) };
+    console.error('[CONNECTION] Database restore failed:', error);
+    
+    // Attempt to rollback if we have a backup and the original database is corrupted
+    if (backupBeforeRestore && fs.existsSync(backupBeforeRestore) && originalDbPath) {
+      try {
+        console.log('[CONNECTION] Attempting to rollback to pre-restore state');
+        if (db && db.open) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)